src/makeInfinityAdder.js

```diff
@@ -5,6 +5,23 @@
  */
 function makeInfinityAdder() {
   // write code here
+  let sum = 0;
+
+  function adder(arg) {
+    if (arg === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    } else {
+      sum += arg;
+
+      return adder;
+    }
+  }
+
+  return adder;
```

Comment:
The function `makeInfinityAdder` is implemented correctly according to the provided task. It defines a closure with a `sum` variable that accumulates the values passed to `adder` until `adder` is called with no arguments. Just ensure that the function is used properly elsewhere to meet the requirement of the task.